weave/trace/vals.py

```diff
@@ -703,11 +703,20 @@ def make_trace_obj(
 
             # need to deref if we encounter these
             if isinstance(val, TableRef):
+                table_row_filter = TableRowFilter()
```

Comment:
Can you write a unit test that hits this path and demonstrates a reduced query?

Reply:
Added